var/spack/repos/builtin/packages/py-mne/package.py

```diff
@@ -11,20 +11,39 @@ class PyMne(PythonPackage):
 
     homepage = "http://mne.tools/"
     pypi = "mne/mne-0.23.4.tar.gz"
+    git = "https://github.com/mne-tools/mne-python.git"
 
+    version("1.2.3", sha256="b300dcee69ffb878cdbc5c02490e877df385c1b9482622e3aa1da06a604a6e37")
+    version("1.2.2", sha256="d40743d6ca7ae3919a557166fd5fc4c00a9719e40c07346baad57964e15f02bb")
     version("0.23.4", sha256="ecace5caacf10961ebb74cc5e0ead4d4dbc55fed006eab1e644da144092354e9")
     version("0.18.2", sha256="aa2e72ad3225efdad39b05e67cd5c88dbd5c3fabf5e1705e459347131f114bc6")
 
-    variant("full", default=False, description="Enable full functionality.")
+    # don't support full variant for newer versions (for now) because dependencies get out of hand
+    variant("full", default=False, when="@:23", description="Enable full functionality.")
+    variant("hdf5", default=False, when="@1:", description="Enable hdf5 functionality.")
 
+    depends_on("python@3.7:", when="@0.24:", type=("build", "run"))
     depends_on("python@3.6:", when="@0.23:", type=("build", "run"))
     depends_on("python@3.5:", type=("build", "run"))
     depends_on("py-setuptools", type="build")
 
+    # requirements_base.txt with versions specified in README.rst (marked with *)
+    depends_on("py-numpy@1.18.1:", when="@1:", type=("build", "run"))  # *
     depends_on("py-numpy@1.15.4:", when="@0.23:", type=("build", "run"))
     depends_on("py-numpy@1.11.3:", type=("build", "run"))
+    depends_on("py-scipy@1.4.1:", when="@1:", type=("build", "run"))  # *
     depends_on("py-scipy@1.1.0:", when="@0.23:", type=("build", "run"))
     depends_on("py-scipy@0.17.1:", type=("build", "run"))
+    depends_on("py-matplotlib@3.1:", when="@1:", type=("build", "run"))  # *
+    depends_on("py-tqdm", when="@1:", type=("build", "run"))
+    depends_on("py-pooch@1.5:", when="@1:", type=("build", "run"))
+    depends_on("py-decorator", when="@1:", type=("build", "run"))
+    depends_on("py-packaging", when="@1:", type=("build", "run"))
+    depends_on("py-jinja2", when="@1:", type=("build", "run"))
+
+    with when("+hdf5"):
+        depends_on("py-h5io", when="@1:", type=("build", "run"))
+        depends_on("py-pymatreader", when="@1:", type=("build", "run"))
```

Comment:
```suggestion
        depends_on("py-h5io", type=("build", "run"))
        depends_on("py-pymatreader", type=("build", "run"))
```
`when` isn't needed since `@:0 +hdf5` isn't possible

Reply:
You're right. I have removed it. Thanks.